### PR TITLE
[ADD]  Historization of the elements related to the EVENT, ACTIVITY, TASK: chatter and audit log

### PR DESCRIPTION
**Pull request by jananjoy**

### CODE REVIEW

**Comment by eilst** [comment]
Check Travis project_event/models/mail_thread.py:88:80: E501 line too long (103 > 79 characters)
project_event/models/mail_thread.py:90:80: E501 line too long (103 > 79 characters)
project_event/models/mail_thread.py:92:80: E501 line too long (103 > 79 characters)
project_event/models/mail_thread.py:106:80: E501 line too long (81 > 79 characters)
project_event/models/mail_thread.py:122:80: E501 line too long (87 > 79 characters)

File: project_event/models/mail_thread.py [hunk 1]
```diff
@@ -57,8 +58,84 @@ def _message_track(self, tracked_fields, initial):
...
+    def order_activity_fields(tracking_values):
+        activity_fields_list = [
+            'name', 'code', 'responsible_id', 'partner_id', 'project_id',
+            'notes', 'category_id', 'date_start', 'date_end', 'task_state',
+            'room_id', 'resource_type', 'manager_id', 'user_id', 'client_type',
+            'sector_id'
+        ]
+        activity_tracking_values = []
+        for index in range(len(tracking_values)):
+            for k in range(len(tracking_values)):
```
Comment (eilst) [inline project_event/models/mail_thread.py:129]: too much nested for loop, isn't there a better way? plus this code seems to be repeated in line 136

**Review by eilst — CHANGES_REQUESTED** [review]

File: project_event/models/mail_thread.py [hunk 2]
```diff
@@ -57,8 +58,84 @@ def _message_track(self, tracked_fields, initial):
                 if tracking:
                     display_values_ids.append([0, 0, tracking])
                     displays.add(col_name)
+        print(tracking_value_ids)
```
Comment (eilst) [inline project_event/models/mail_thread.py]:
```suggestion
```

File: project_event/models/mail_thread.py (hunk 1 above)
Comment (jananjoy) [reply project_event/models/mail_thread.py:129]: Plz  suggest four  optimized code.

**Comment by eilst** [comment]
It is good :+1:, but the def get_type(tracking_values): is not the best. Better to use polymorphism. So I created two PRs, that contain the code of this PR without the def get_type(tracking_values):

**Review by eilst — APPROVED** [review]

**Comment by eilst** [comment]
https://github.com/savoirfairelinux/project-addons/pull/459 and https://github.com/savoirfairelinux/project-addons/pull/460

**Pull request closed without merging**